airflow/exceptions.py

```diff
@@ -207,6 +207,22 @@ def __init__(self, *args, **kwargs):
         warnings.warn("DagFileExists is deprecated and will be removed.", DeprecationWarning, stacklevel=2)
 
 
+class DagInvalidTriggerRule(AirflowException):
+    """Raise when a dag has 'fail_stop' enabled yet has a non-default trigger rule"""
+
+    @staticmethod
+    def check(dag: DAG | None, trigger_rule: str):
+        from airflow.models.abstractoperator import DEFAULT_TRIGGER_RULE
+
+        if dag is not None and dag.fail_stop and trigger_rule != DEFAULT_TRIGGER_RULE:
+            raise DagInvalidTriggerRule()
```

Comment:
Seems simpler if this is a classmethod and use `raise cls`?

Reply:
Good point. I have updated the code to be a class method rather than a static method.